pkg/controllers/serviceca/service_ca_controller.go

```diff
@@ -0,0 +1,223 @@
+	c := &serviceCAController{
+		services:       services,
+		secrets:        secrets,
+		configMaps:     configMaps,
+		operatorClient: operatorClient,
+	}
+	return factory.New().WithInformers(
+		kubeInformersForTargetNamespace.Core().V1().Secrets().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().Services().Informer(),
+		kubeInformersForTargetNamespace.Core().V1().ConfigMaps().Informer(),
+		configInformer.Config().V1().Authentications().Informer(),
+		configInformer.Config().V1().Ingresses().Informer(),
+	).ResyncEvery(30*time.Second).WithSync(c.sync).ToController("ServiceCAController", recorder.WithComponentSuffix("service-ca-controller"))
+}
+
+func (c *serviceCAController) sync(ctx context.Context, syncCtx factory.SyncContext) error {
+	foundConditions := []operatorv1.OperatorCondition{}
+
+	// make sure we create the service before we start asking about service certs
+	_, _, err := resourceapply.ApplyService(c.services, syncCtx.Recorder(), getOAuthService())
+	if err != nil {
+		foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+			Type:    "OAuthServiceDegraded",
+			Status:  operatorv1.ConditionTrue,
+			Reason:  "ApplyFailed",
+			Message: fmt.Sprintf("Failed to apply %q: %v", "oauth-openshift", err),
+		})
+	}
+
+	if len(foundConditions) == 0 {
+		serviceCAConditions, err := c.getServiceCA(ctx, syncCtx.Recorder())
+		if err != nil {
+			return err
+		}
+		foundConditions = append(foundConditions, serviceCAConditions...)
+	}
+
+	// TODO: Remove this as soon as we break the ordering of the main operator
+	if len(foundConditions) == 0 {
+		foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+			Type:   "AuthServiceCAProgressing",
+			Status: operatorv1.ConditionFalse,
+			Reason: "AsExpected",
+		})
+	} else {
+		foundConditions = append(foundConditions, operatorv1.OperatorCondition{
+			Type:   "AuthServiceCAProgressing",
+			Status: operatorv1.ConditionTrue,
+			Reason: fmt.Sprintf("%d degraded conditions found while working towards service ca", len(foundConditions)),
+		})
+	}
+
+	updateConditionFuncs := []v1helpers.UpdateStatusFunc{}
+
+	for _, conditionType := range knownConditionNames.List() {
+		// clean up existing foundConditions
+		updatedCondition := operatorv1.OperatorCondition{
+			Type:   conditionType,
+			Status: operatorv1.ConditionFalse,
+		}
+		if condition := v1helpers.FindOperatorCondition(foundConditions, conditionType); condition != nil {
+			updatedCondition = *condition
+		}
+		updateConditionFuncs = append(updateConditionFuncs, v1helpers.UpdateConditionFn(updatedCondition))
+	}
+
+	if _, _, err := v1helpers.UpdateStatus(c.operatorClient, updateConditionFuncs...); err != nil {
+		return err
+	}
+
+	// retry faster if we have degraded condition
+	// TODO: replace this with len(foundConditions) != 0 after all controllers are extracted.
+	if v1helpers.IsOperatorConditionTrue(foundConditions, "AuthServiceCAProgressing") {
+		return factory.SyntheticRequeueError
+	}
```

Comment:
if you're to re-use this pattern, you should extract it somewhere

Reply:
already noticed that, again a follow up :-)

Reply:
(it is part of "part 3" ;-)